### PR TITLE
fix: add observation for sdcore management relation joined

### PR DESCRIPTION
**Pull request by gatici**
# Description

1.4/edge  branch uses latest revision of nms image.
While running E2E sdcore-tests sometimes NMS stuck at `installing agent` status and  `sdcore-management-relation-changed` hook fails.The reason of this failure is charm does not include storage paths of UPF and GNB config. As they are moved to webui which provides endpoints to configure UPF and GNB.
These are the related changes which causes the issue.

- https://github.com/canonical/sdcore-nms/pull/621
- https://github.com/omec-project/webconsole/pull/194

As a solution UPF and GNB config paths are removed from the charm. 

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -57,6 +57,9 @@ def __init__(self, *args):
         self.framework.observe(self.on.nms_pebble_ready, self._configure_sdcore_nms)
         self.framework.observe(self.on.update_status, self._configure_sdcore_nms)
         self.framework.observe(self.fiveg_n4.on.fiveg_n4_available, self._configure_sdcore_nms)
+        self.framework.observe(
+            self.on.sdcore_management_relation_joined, self._configure_sdcore_nms
```
Comment (gruyaume) [inline src/charm.py:61]: How could this have caused the problem? We already observe `secore_management_url_available` which would be triggered after the relation joined.

Comment (gatici) [reply src/charm.py:61]: Right, I also found this strange.  We got following error in the logs:
```
unit-nms-0: 2024-09-23 08:59:15 ERROR juju.worker.uniter.operation hook "sdcore-management-relation-changed" (via hook dispatching script: dispatch) failed: exit status 1
unit-nms-0: 2024-09-23 08:59:15 DEBUG juju.machinelock created rotating log file "/var/log/juju/machine-lock.log" with max size 10 MB and max backups 5
unit-nms-0: 2024-09-23 08:59:15 DEBUG juju.machinelock machine lock "machine-lock" released for nms/0 uniter (run relation-changed (20; app: webui) hook)
unit-nms-0: 2024-09-23 08:59:15 DEBUG juju.worker.uniter.operation lock released for nms/0
unit-nms-0: 2024-09-23 08:59:15 DEBUG juju.worker.uniter.actions no next action from pending=[]; completed=map[]
unit-nms-0: 2024-09-23 08:59:15 INFO juju.worker.uniter awaiting error resolution for "relation-changed" hook
unit-nms-0: 2024-09-23 08:59:15 DEBUG juju.worker.uniter [AGENT-STATUS] error: hook failed: "sdcore-management-relation-changed"
```

Comment (gatici) [reply src/charm.py]: The real reason is found which causes failure in sdcore-management-relation changes. 
NMS image does not include GNB and UPF config paths any more as described in the PR definition.

**Comment by gatici** [comment]
This change does not solve the problem as NMS can not start independently using current image https://github.com/canonical/sdcore-nms/blob/main/rockcraft.yam. The image is designed to run only Webui service.

**Pull request closed without merging**